tests_integration/nativeapp/test_version.py

```diff
@@ -48,29 +48,52 @@ def normalize_identifier(identifier: Union[str, int]) -> str:
         return unquote_identifier(to_identifier(id_str))
 
 
+@pytest.fixture(params=["napp_init_v1", "napp_init_v2"])
+def _test_project(request):
+    return request.param
+
+
+@pytest.fixture(params=["app version create", "ws version create --entity-id=pkg"])
+def _create_command(request):
+    return request.param
+
+
+@pytest.fixture(params=["app version list", "ws version list --entity-id=pkg"])
+def _list_command(request):
+    return request.param
+
+
+@pytest.fixture(params=["app version drop", "ws version drop --entity-id=pkg"])
+def _drop_command(request):
+    return request.param
+
+
+@pytest.fixture()
+def command_parametrization(
+    _create_command, _list_command, _drop_command, _test_project
+):
+    if "v1" in _test_project and (
+        "ws" in " ".join([_create_command, _list_command, _drop_command])
+    ):
+        pytest.skip("ws commands are not supported on v1 projects")
+    return _create_command, _list_command, _drop_command, _test_project
```

Comment:
oof, maybe I'm just tired, but I find it hard to wrap my head around the combinations being assembled here. Does that mean we'll test all permutations of each of the 4 params?

Reply:
this fixture requests 4 fixtures, each of which is parametrized to return both the `snow app` and `snow ws` version of the command, for a total of 16 distinct test cases. Since `snow ws` can't operate on v1 projects, we skip the test in all the permutations where one of the commands is `snow ws`. In practice, this results in 9 test cases, 8 for v2 projects and 1 for a v1 project _and_ where the command sequence goes `snow app version create`, `snow app version list`, `snow app version drop`.

Reply:
Not a fan of this pattern TBH, IMHO it makes it really tricky to grok what's being covered and what might be missing. I see this is merged already. I'd prefer not reusing this pattern elsewhere.

Reply:
Yeah sorry, forgot to disable the auto-merge. I'm gonna remove this when I do my test parametrization pruning.